Weapons

sci10n opened the pull request:
Removed sprite rendering but implemented weapon functionality with entity projectiles.

Test by running the LevelSandbox

sci10n committed:
added request queue for inter entity comunications

sci10n committed:
added request queue

sci10n committed:
changed generic type of request queue and added weapons to entities

sci10n committed:
reverted to dynamic bodies for players and bots

sci10n committed:
entitity and body cleanup when projectile hits objects

sci10n committed:
code cleanup and refractoring of physics/collision resover systems

CaffeineViking commented:
I haven't looked through everything, but what I have looked through looks pretty good. I've removed the need for the multi-case for now since Gradle is complaining that we are using source compatibility of 1.6 while multi-case is source compatibility 1.7 (it seems).

CaffeineViking committed:
Temporarily removed need to have multi-case catch.

We either need to remove the restriction of having source compatibility
set to 1.6 in Gradle (in the build settings) or not use multicase catch.
For now I've removed the multi-case, but we might want that later, so we
can discuss what to do. Or maybe I'm just doing something wrong, I dunno.
